cmd/network/verification.go

```diff
@@ -580,22 +601,29 @@ func (e *EgressVerification) setupForPodMode(ctx context.Context) (*onvKubeClien
 	var restConfig *rest.Config
 	var err error
 
-	// Prefer backplane credentials when cluster ID is available
-	if e.ClusterId != "" {
+	if _, err := os.Stat("/var/run/secrets/kubernetes.io/serviceaccount/token"); err == nil {
+		restConfig, err = rest.InClusterConfig()
+		if err == nil {
+			e.log.Info(ctx, "Pod mode using in-cluster configuration with ServiceAccount")
+		} else {
+			e.log.Info(ctx, "ServiceAccount token found but in-cluster config failed, falling back to other methods")
```

Comment:
I think the priority should be a bit different here:

1. If I pass `--kubeconfig`, I explicitly want to use that always
2. If I pass `--cluster-id`, I want to use backplane
3. If I don't pass `--kubeconfig` or `--cluster-id`, try in-cluster
4. If in cluster fails, fall back to default kubeconfig

I would suggest moving this decision logic into a simple function so it reads easier within `setupForPodMode`.